### DISCUSSION TITLE
labgrid.protocol.CommandProtocol run_check

### DISCUSSION THREAD

**Discussion by AndreasWilligWES** [labgrid-project/labgrid#1379] [Q&A] [answered]
Hi all,
i am using a wrapper fixture containing CommandProtocol (instanced with ssh driver) and issue a run_check.
Point is, i have to decide on call return code how to proceed the testing. I did check the source and callstack of the run_check call, but did not see whether/where the exitcode of the remote command is stored (call gives it into stdout,stderr,exitcode and the ExecError Exception does not get the exitcode as parm).
Do i miss the forest but the trees or is there a way to get the exitcode?

And yes i know: basic usage is not expecting anything but valid or stop, in my case the customer likes to do some provisioning during test, which requires different wait times (longer than regular ssh timeout) depending whether some settings have to be updated or are already correct.

I'd like to do sth like (simple pseudo code):
```python
try:
   cli.run_check(___provision.sth)
except ExecutionError as e:
   if e.retcode = 0x1003:
      time.sleep(30)
      cli.run_check(__verifyprovision)
   else:
      raise e
test_device()
``` 

Any ideas are welcome, i do not want to modify sshdriver if not absolutely required, i guess someone had sth like that before?

Regards

Andreas

**Comment by Emantor** [comment, marked as answer]
You'll want to use `run()` instead of `run_check()`. `run()` returns the return code and your script can decide how to proceed.
With your pseudo code:
```python
_, _, retcode = cli.run(___provision.sth)

if retcode = 0x1003:
    time.sleep(30)
    cli.run_check(__verifyprovision)
test_device()
```

**Reply by AndreasWilligWES** [reply]
I'll try, sounds like a reasonable solution, thanks
